tensorly/tests/test_tr_tensor.py

```diff
@@ -46,24 +46,10 @@ def test_tr_to_tensor():
     factors = [tl.randn((2, 4, 3)), tl.randn((3, 5, 2)), tl.randn((2, 6, 2))]
 
     # Create tensor
-    tensor = tl.zeros((4, 5, 6))
-
-    for i in range(4):
-        for j in range(5):
-            for k in range(6):
-                product = tl.dot(
-                    tl.dot(factors[0][:, i, :], factors[1][:, j, :]),
-                    factors[2][:, k, :],
-                )
-                # TODO: add trace to backend instead of this
-                tensor = tl.index_update(
-                    tensor,
-                    tl.index[i, j, k],
-                    tl.sum(product * tl.eye(product.shape[0])),
-                )
+    tensor = tl.einsum("iaj,jbk,kci->abc", *factors)
```

Comment:
We have trace in the backend now (see TODO comment), but I thought this was cleaner.